### PR TITLE
restore points sdk

### PR DESCRIPTION
**Pull request by sarvesh-s**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by sarvesh-s** [comment]
Rest spec PR, https://github.com/Azure/azure-rest-api-specs-pr/pull/374

**Comment by shahabhijeet** [comment]
@sarvesh-s seems there are breaking changes, if yes then you will have bump up the version for non-preview release.
Also I will need a sign off from Jared Moore on this PR

**Comment by jaredmoo** [comment]
RestorePoints has been renamed to DatabaseRestorePoints. Can you undo this by changing DatabaseRestorePoints_* to RestorePoints_* in swagger?

**Comment by sarvesh-s** [comment]
Taken care of backward compatibility

File: src/SDKs/SqlManagement/Management.Sql/Generated/IRestorePointsOperations.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Management.Sql/Generated/IRestorePointsOperations.cs]: Sounds like the grammar is not quire right here. Can you change it in Swagger back to the original (`Gets a list of database restore points.`)

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: use camelcase: `getResponse`

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: DatabaseEdition.DataWarehouse

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: ServiceObjectiveName.DW100

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: Asserts on the HttpStatusCode are not needed because the client is already doing that internally. So you can write this instead:

```
                RestorePoint postResponse = sqlClient.RestorePoints.Create(
                        resourceGroup.Name,
                        server.Name,
                        db.Name,
                        restoreDefinition);
```

similar for below

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: Need to validate the properties of `postResponse`

Comment (jaredmoo) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: Similar for below

**Review by jaredmoo — CHANGES_REQUESTED** [review]

File: src/SDKs/SqlManagement/Management.Sql/Generated/IRestorePointsOperations.cs
Comment (sarvesh-s) [reply src/SDKs/SqlManagement/Management.Sql/Generated/IRestorePointsOperations.cs]: https://github.com/Azure/azure-rest-api-specs/pull/2477

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (sarvesh-s) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: Fixed, will do label validation next backend train deployment. Currently validating name only.

**Comment by jaredmoo** [comment]
I mean validate in the SDK test that the returned objects have expected property values

File: src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs
Comment (sarvesh-s) [reply src/SDKs/SqlManagement/Sql.Tests/DatabaseRestoreScenarioTests.cs]: Added.

**Review by shahabhijeet — APPROVED** [review]
This is good to go as soon as @jaredmoo approves it.

**Review by jaredmoo — CHANGES_REQUESTED** [review]
Needs to be put on hold due to reliability issues in the backend that will cause re-recording to be painful.

**Comment by shahabhijeet** [comment]
@sarvesh-s if this needs to be put on hold. Can you re-open this PR when you are ready or create a new PR.
This PR will be closed EOW if not ready to be merged.

**Comment by shahabhijeet** [comment]
Please reopen when this is ready to be merged

**Pull request closed without merging**
